indico/modules/rb_new/client/js/common/rooms/EquipmentList.jsx

```diff
@@ -0,0 +1,108 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import connect from "react-redux/es/connect/connect";
+import _ from "lodash";
+import * as roomsSelectors from "./selectors";
+
+import {Dropdown, Icon, List} from "semantic-ui-react";
+
+import './RoomEditModal.module.scss';
+
+
+class EquipmentList extends React.Component{
+    static propTypes = {
+        onChange: PropTypes.func.isRequired,
+        onFocus: PropTypes.func.isRequired,
+        onBlur: PropTypes.func.isRequired,
+        value: PropTypes.array.isRequired,
+        label: PropTypes.string.isRequired,
+        disabled: PropTypes.bool,
+        equipmentTypes: PropTypes.arrayOf(PropTypes.object).isRequired,
+    };
+
+    static defaultProps = {
+        disabled: false,
+    };
+
+    generateEquipmentOptions = () => {
+        const {value} = this.props;
+        const {equipmentTypes} = this.props;
+        const options = [];
+        equipmentTypes.map((equipmentType) => (value.indexOf(equipmentType.id) < 0
+            ? options.push({key: equipmentType.id, text: equipmentType.name, value: equipmentType.id}) : null));
+        return options;
+    };
+
+    render() {
+        const {onChange, onFocus, onBlur, value, disabled, equipmentTypes, label} = this.props;
+        if (!equipmentTypes || !value) {
```

Comment:
when would `equipmentTypes` be falsy? isn't it rather an empty list if there are none available?

Reply:
same for the value i guess